packages/beacon-node/src/api/impl/utils.ts

```diff
@@ -0,0 +1,25 @@
+import {ApiError} from "./errors.js";
+
+/**
+ * Ensures that the array contains unique values, and throws an ApiError
+ * otherwise.
+ * @param array - The array to check for uniqueness.
+ * @param message - The message to put in the ApiError if the array contains
+ * duplicates.
+ */
+export function ensureUniqueItemsOrThrow(array: unknown[] | undefined, message: string): void {
+  if (!array) {
+    return;
+  }
+
+  const duplicateItems = array.reduce((partialDuplicateItems: unknown[], item, index) => {
+    if (array.indexOf(item) !== index && !partialDuplicateItems.includes(item)) {
+      return partialDuplicateItems.concat(item);
+    }
+    return partialDuplicateItems;
+  }, []);
+
+  if (duplicateItems.length) {
+    throw new ApiError(400, `${message} (Duplicate Items: ${duplicateItems.join(", ")})`);
```

Comment:
message seems redundant, what about

```suggestion
    throw new ApiError(400, `${message}: ${duplicateItems.join(", ")}`);
```